Source/towners.cpp

```diff
@@ -898,6 +898,12 @@ bool DebugTalkToTowner(std::string targetName)
 	std::transform(targetName.begin(), targetName.end(), targetName.begin(), [](unsigned char c) { return std::tolower(c); });
 	auto &myPlayer = Players[MyPlayerId];
 	for (auto &towner : TownerInitList) {
+		// prevent going into hellfire territory without hellfire
+		if (towner.type == TOWN_COWFARM && !gbIsHellfire)
+			return false;

```

Comment:
`InitTowners` has a similar check that is even better (checks also quest state).
For example now you could talk to girl but theo quest is disabled.

Reply:
Meh, copying that check would duplicate a lot of code and I can't just call that function because it modifies dMonsters

Reply:
Can't [these lines](https://github.com/diasurgical/devilutionX/blob/master/Source/towners.cpp#L819-L838) be refactored in a method `IsTownerPresent` that returns a bool?

Reply:
I guess :)